nodejs/eks/cluster.ts

```diff
@@ -184,38 +189,53 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
     // Create the VPC CNI management resource.
     const vpcCni = new VpcCni(`${name}-vpc-cni`, kubeconfig, args.vpcCniOptions, { parent: parent });
 
-    // Create the instance role we'll use for worker nodes.
-    let instanceRole: pulumi.Output<aws.iam.Role>;
-    if (args.instanceRole) {
-        instanceRole = pulumi.output(args.instanceRole);
+    const instanceRoleMappings: RoleMapping[] = [];
+    let instanceProfile: aws.iam.InstanceProfile;
+    if (args.instanceRoleArns && args.instanceRoleArns.length) {
+        // Create role mappings if an array of instanceRoleArn's is specified.
+        for (const [index, arn] of args.instanceRoleArns.entries()) {
+            const roleName = pulumi.output(arn).apply(r => r.split("/")).apply(r => r[1]);
+            const role = aws.iam.Role.get(`${name}-existingRole-${index}`, roleName);
+
+            // Add the EKS IAM Policies to the roles provided to join the cluster.
+            addPoliciesToExistingRole(`${name}-eksServicePolicies-role-${index}`, role, eksManagedPolicyArns);
+
+            // Create a role mapping of the role for aws-auth.
+            instanceRoleMappings.push(createInstanceRoleMapping(arn));
+        }
     } else {
-        instanceRole = (new ServiceRole(`${name}-instanceRole`, {
-            service: "ec2.amazonaws.com",
-            managedPolicyArns: [
-                "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
-                "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
-                "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
-            ],
-        }, { parent: parent })).role;
-    }
-    const instanceRoleARN = instanceRole.apply(r => r.arn);
+        // Create the instance role we'll use for worker nodes.
+        let instanceRole: pulumi.Output<aws.iam.Role>;
+        if (args.instanceRole) {
+            instanceRole = pulumi.output(args.instanceRole);
+        } else {
+            instanceRole = (new ServiceRole(`${name}-instanceRole`, {
+                service: "ec2.amazonaws.com",
+                managedPolicyArns: eksManagedPolicyArns,
+            }, { parent: parent })).role;
+        }
+        const instanceRoleArn = instanceRole.apply(r => r.arn);
+
+        // Create a new policy for the role, if specified.
+        if (args.customInstanceRolePolicy) {
```

Comment:
We could choose if we want to proactively mark this argument as "deprecated" and point to the ability to pass `instanceRoles` instead?